apps/andi/lib/andi/event_handler.ex

```diff
@@ -15,7 +15,15 @@ defmodule Andi.EventHandler do
   @ingested_time_topic "ingested_time_topic"
 
   def handle_event(%Brook.Event{type: dataset_update(), data: %Dataset{} = data}) do
-    DatasetCache.put(data)
+    dataset =
+      case Andi.Repo.get(Andi.InputSchemas.Dataset, data.id) do
```

Comment:
we'll be putting this under the repo pattern eventually. Something like `Andi.Schemas.Datasets.get_or_create`